tests/Unit/Domain/Test_AreaElement.cpp

```diff
@@ -0,0 +1,230 @@
+// corresponds to the expected area
+void test_kerr_area() {
+  for (const auto& [ref_level, mass, spin] : cartesian_product(
+           std::array<size_t, 3>{{0, 1}}, make_array(0.001, 1., 100.),
+           std::array<std::array<double, 3>, 3>{{make_array(0., 0., 0.),
+                                                 make_array(0.1, 0.2, 0.3),
+                                                 make_array(0., 0., 0.99)}})) {
+    const double horizon_radius = mass * (1 + sqrt(1 - dot(spin, spin)));
+    domain::creators::Sphere shell{horizon_radius,
+                                   10. * horizon_radius,
+                                   domain::creators::Sphere::Excision{},
+                                   ref_level,
+                                   size_t(10),
+                                   true};
+    const auto shell_domain = shell.create_domain();
+    const auto& blocks = shell_domain.blocks();
+    const auto& initial_ref_levels = shell.initial_refinement_levels();
+    const auto element_ids = initial_element_ids(initial_ref_levels);
+    const auto excision_sphere =
+        shell_domain.excision_spheres().at("ExcisionSphere");
+    const Mesh<2> face_mesh{10, Spectral::Basis::Legendre,
+                            Spectral::Quadrature::GaussLobatto};
+    domain::CoordinateMaps::KerrHorizonConforming kerr_schild_map(mass, spin);
+
+    // Shell does not distort to conform to the horizon in Kerr-Schild
+    // coordinates for spinning black holes, so we manually append the horizon
+    // conforming map
+    const auto grid_to_kerr_conforming_map =
+        CoordinateMap<Frame::Inertial, Frame::Inertial,
+                      domain::CoordinateMaps::KerrHorizonConforming>(
+            kerr_schild_map);
+    gr::Solutions::KerrSchild kerr_schild{mass, spin, {0., 0., 0.}};
+    double horizon_area = 0.;
+    for (const auto& element_id : element_ids) {
+      const auto& current_block = blocks.at(element_id.block_id());
+      const auto element_abutting_direction =
+          excision_sphere.abutting_direction(element_id);
+      if (element_abutting_direction.has_value()) {
+        const auto face_logical_coords = interface_logical_coordinates(
+            face_mesh, element_abutting_direction.value());
+        const ElementMap logical_to_grid_map(
+            element_id, current_block.stationary_map().get_clone());
+        const auto inertial_coords = logical_to_grid_map(face_logical_coords);
+        const auto inv_jacobian =
+            logical_to_grid_map.inv_jacobian(face_logical_coords);
+        const auto inv_jacobian_kerr_horizon =
+            grid_to_kerr_conforming_map.inv_jacobian(inertial_coords);
+        const auto total_inv_jacobian = tenex::evaluate<ti::I, ti::j>(
+            inv_jacobian(ti::I, ti::k) *
+            inv_jacobian_kerr_horizon(ti::K, ti::j));
+        const auto spacetime_vars = kerr_schild.variables(
+            grid_to_kerr_conforming_map(inertial_coords), 0.,
```

Comment:
I think the different maps and frames are a bit confusing here (e.g. map is named grid but coords are named inertial, then there's another Inertial->Inertial map). Please clear this up a bit, or better yet use a Composition map that combines ElementLogical->BlockLogical->Wedge->HorizonConforming into one map (see my comment above https://github.com/sxs-collaboration/spectre/pull/4743#discussion_r1134797054).